### ISSUE TITLE
[CI] SLMSnapshotBlockingIntegTests.testRetentionWhileSnapshotInProgress failing

### ISSUE THREAD

**Issue by mark-vieira** [elastic/elasticsearch#47834]
`org.elasticsearch.xpack.slm.SLMSnapshotBlockingIntegTests.testRetentionWhileSnapshotInProgress` has started failing pretty often now with the following error:

```
java.lang.AssertionError: expected at least one master-eligible node left in {node_sc1=org.elasticsearch.test.InternalTestCluster$NodeAndClient@6e458321}
```

Here are some example build scans:
https://gradle-enterprise.elastic.co/s/hlldgfw3cx4bc/tests/rfsroxnx4sflo-uvkkyo6qd6mno
https://gradle-enterprise.elastic.co/s/i4ncjbmiflt6a/tests/rfsroxnx4sflo-uvkkyo6qd6mno
https://gradle-enterprise.elastic.co/s/33mp5rqowzsbi/tests/rfsroxnx4sflo-uvkkyo6qd6mno

Looking at [build-stats](https://build-stats.elastic.co/app/kibana#/discover?_g=(refreshInterval:(pause:!t,value:0),time:(from:now%2Fw,mode:quick,to:now%2Fw))&_a=(columns:!(branch,message),index:e58bf320-7efd-11e8-bf69-63c8ef516157,interval:auto,query:(language:kuery,query:'test:%20%22testRetentionWhileSnapshotInProgress%22'),sort:!(time,desc))) this looks to be happening in both `master` and `7.x`.

**Comment by mark-vieira** [comment]
This has been failing often enough that I've muted this in `master` and `7.x`.

**Comment by mayya-sharipova** [comment]
The test failed on the intake: https://gradle-enterprise.elastic.co/s/anqtao57gamow
```
org.elasticsearch.repositories.RepositoryException: [my-repo] Could not determine repository generation from root blobsClose stacktrace
at __randomizedtesting.SeedInfo.seed([AF89D7C3B10D5638:D253C9D96B797893]:0)
at org.elasticsearch.repositories.blobstore.BlobStoreRepository.getRepositoryData(BlobStoreRepository.java:906)
at org.elasticsearch.snapshots.SnapshotsService.getRepositoryData(SnapshotsService.java:163)
at org.elasticsearch.action.admin.cluster.snapshots.status.TransportSnapshotsStatusAction.buildResponse(TransportSnapshotsStatusAction.java:201)
at org.elasticsearch.action.admin.cluster.snapshots.status.TransportSnapshotsStatusAction.masterOperation(TransportSnapshotsStatusAction.java:105)
at org.elasticsearch.action.admin.cluster.snapshots.status.TransportSnapshotsStatusAction.masterOperation(TransportSnapshotsStatusAction.java:65)
at org.elasticsearch.action.support.master.TransportMasterNodeAction$AsyncSingleAction.lambda$doStart$3(TransportMasterNodeAction.java:166)
at org.elasticsearch.action.ActionRunnable$2.doRun(ActionRunnable.java:73)
at org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingAbstractRunnable.doRun(ThreadContext.java:769)
at org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:37)
at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
at java.lang.Thread.run(Thread.java:834)
Caused by: java.nio.file.NoSuchFileException: /dev/shm/elastic+elasticsearch+master+multijob+fast+part2/x-pack/plugin/ilm/build/testrun/test/temp/org.elasticsearch.xpack.slm.SLMSnapshotBlockingIntegTests_AF89D7C3B10D5638-001/tempDir-002/repos/UOoOkN/index-0Open stacktrace
[2019-10-23T23:58:06,325][INFO ][o.e.x.s.SLMSnapshotBlockingIntegTests] [testRetentionWhileSnapshotInProgress] before test
[2019-10-23T23:58:06,341][INFO ][o.e.x.s.SLMSnapshotBlockingIntegTests] [testRetentionWhileSnapshotInProgress] [SLMSnapshotBlockingIntegTests#testRetentionWhileSnapshotInProgress]: setting up test
[2019-10-23T23:58:06,416][INFO ][o.e.t.InternalTestCluster] [testRetentionWhileSnapshotInProgress] Setup InternalTestCluster [TEST-TEST_WORKER_VM=[72]-CLUSTER_SEED=[-2565761215536046425]-HASH=[3B7F3AA48D]-cluster] with seed [DC6495C4B97FDEA7] using [0] dedicated masters, [0] (data) nodes and [0] coord only nodes (master nodes are [auto-managed])
[2019-10-23T23:58:06,936][INFO ][o.e.x.s.SLMSnapshotBlockingIntegTests] [testRetentionWhileSnapshotInProgress] [SLMSnapshotBlockingIntegTests#testRetentionWhileSnapshotInProgress]: all set up test
[2019-10-23T23:58:06,938][INFO ][o.e.x.s.SLMSnapshotBlockingIntegTests] [testRetentionWhileSnapshotInProgress] --> starting enough nodes to ensure we have enough to safely stop for tests
```

**Comment by original-brownbear** [comment]
This will be trivial to fix now thanks to #48329 , I'm on it :)

**Comment by dnhatn** [comment]
```
org.elasticsearch.client.SnapshotIT > testCreateSnapshot FAILED
    org.elasticsearch.ElasticsearchStatusException: Elasticsearch exception [type=repository_exception, reason=[test_repository] Could not determine repository generation from root blobs]
        at __randomizedtesting.SeedInfo.seed([BCF37B148E73CF08:4295EEA7B8C03D11]:0)
        at org.elasticsearch.rest.BytesRestResponse.errorFromXContent(BytesRestResponse.java:177)
        at org.elasticsearch.client.RestHighLevelClient.parseEntity(RestHighLevelClient.java:1793)
        at org.elasticsearch.client.RestHighLevelClient.parseResponseException(RestHighLevelClient.java:1770)
        at org.elasticsearch.client.RestHighLevelClient.internalPerformRequest(RestHighLevelClient.java:1527)
        at org.elasticsearch.client.RestHighLevelClient.performRequest(RestHighLevelClient.java:1484)
        at org.elasticsearch.client.RestHighLevelClient.performRequestAndParseEntity(RestHighLevelClient.java:1454)
        at org.elasticsearch.client.SnapshotClient.delete(SnapshotClient.java:344)
        at org.elasticsearch.client.ESRestHighLevelClientTestCase.execute(ESRestHighLevelClientTestCase.java:90)
        at org.elasticsearch.client.ESRestHighLevelClientTestCase.execute(ESRestHighLevelClientTestCase.java:81)
        at org.elasticsearch.client.SnapshotIT.testCreateSnapshot(SnapshotIT.java:167)

        Caused by:
        org.elasticsearch.ElasticsearchException: Elasticsearch exception [type=no_such_file_exception, reason=/dev/shm/elastic+elasticsearch+master+multijob+fast+part1/client/rest-high-level/build/testclusters/integTest-0/repo/index-18]
            at org.elasticsearch.ElasticsearchException.innerFromXContent(ElasticsearchException.java:496)
            at org.elasticsearch.ElasticsearchException.fromXContent(ElasticsearchException.java:407)
            at org.elasticsearch.ElasticsearchException.innerFromXContent(ElasticsearchException.java:437)
            at org.elasticsearch.ElasticsearchException.failureFromXContent(ElasticsearchException.java:603)
            at org.elasticsearch.rest.BytesRestResponse.errorFromXContent(BytesRestResponse.java:169)
            ... 9 more
REPRODUCE WITH: ./gradlew ':client:rest-high-level:integTestRunner' --tests "org.elasticsearch.client.SnapshotIT.testCreateSnapshot" -Dtests.seed=BCF37B148E73CF08 -Dtests.security.manager=true -Dtests.locale=es-US -Dtests.timezone=America/Jujuy -Dcompiler.java=12 -Druntime.java=11
```
This failed again on master but with HLRC: https://gradle-enterprise.elastic.co/s/5abbkuha6acvy/console-log
